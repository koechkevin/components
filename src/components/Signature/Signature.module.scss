@import '../../styles/global.scss';


.label {
  &:global(.ant-tag) {
    display: flex;
    align-items: center;
    font-weight: normal;
    text-align: center;
    border: none;
    background: transparent;
    margin-left: -7px;
  }

  .outerLabel {
    margin-right: 7px;
    font-size: 12px;
    font-weight: 600;
    height: 22px;
    line-height: 22px;
    font-weight: normal;
    color: $color-hex-gray-light;
  }

}


.default {
  display: flex;
  justify-content: center;
  align-items: center;
  background: $color-primary-light;
  color: $color-primary;
  font-weight: 600;
  font-size: 12px;
  border: none;
}

.avatar {
  color: $color-avatar-orange;
  background-color: $color-avatar-light;
}