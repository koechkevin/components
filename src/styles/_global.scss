// VARIABLES
// ==========================================================================
// COLOR VARIABLES
$color-white: #fff;
$color-black: #000;
$color-black-light: #272727;
$color-black-lighter: #ebebeb;
$color-dark-lighter: #2c2c2e;

$color-primary: #0050c8;
$color-primary-light: #dfe9f8;
$color-primary-dark: #003c96;

$color-success: #39c049;
$color-success-secondary: #2b9037;
$color-success-light: #e6f7e8;

$color-error: #c81400;
$color-error-secondary: #ff4b33;
$color-error-hover-secondary: #e6432e;
$color-red-secondary: #fa2d19;
$color-error-hover: #960f00;
$color-light-red: #f8e2df;

// ------------------------------------------------ Base Dark
$color-base-dark: #1d1d1d;
$color-base-darker: #161616;
$color-side-bar-dark: #282828;
$color-base-dark-gray: #565656;
$color-base-gray-dark: #6b6b6b;
$color-base-gray-darker: #eaeaea;
$color-base-gray-light: #b9b9b9;
$color-base-gray: #c7c7c7;
$color-dark-gray: #2e2f30;
$color-base-gray-lighter: #cbcbcb;
$color-button-gray: #eaeaea;
$color-grey-light: #979797;
$color-lighter-dark: #e4e4e4;
$color-grey-lighter: #d2d2d2;

$color-gray-regular: #ebebeb;
$color-gray-dark: #d0d0d0;
$color-gray-darker: #606060;
$color-blue-gray: #f8fbfd;
$color-darker-blue-gray: #f2f8fd;
$color-text-gray: #707070;
$color-basic-light-gray: #f3f3f3;
$color-lighter-gray: #d8d8d8;
$color-light-gray: #d9d9d9;

$color-buttercup: #fb6d77;
$color-avatar-orange: #f56a00;
$color-light-orange: #ecb22e;
$color-avatar-light: #fde3cf;

$color-font-dark: #555;
$color-font-darker: #333;

$color-gray-light: #f8f8f8;
$color-gray-lighter: #fafafa;

$color-border-gray: #cdcfd3;
$color-hover-bg: #e1e9f7;
$color-danger-bg: #f4e3e0;
$color-hex-gray-primary: #545454;
$color-hex-gray-light: #7b7b7b;

// RADIO Colors
$color-purple-radio: #9c43d6;
$color-purple-radio-light: #eedff8;
$color-gold-radio: #d7a84b;
$color-gold-radio-light: #faeccb;
$color-radio-green: #2b7f35;
$color-green-success: #009426;
$color-radio-green-light: #dfece1;

// RESPONSIVE
$breakpoints: (
  'lg-large': (
    max-width: 1440px,
  ),
  'lg-medium': (
    max-width: 1320px,
  ),
  'lg-small': (
    max-width: 1280px,
  ),
  'lg': (
    max-width: 1200px,
  ),
  'md-large': (
    max-width: 1024px,
  ),
  'md': (
    max-width: 992px,
  ),
  'sm': (
    max-width: 768px,
  ),
  'xs': (
    max-width: 320px,
  ),
  'xs-m': (
    max-width: 375px,
  ),
  'xs-l': (
    max-width: 575px,
  ),
  'xs-l-landscape': (
    min-width: 320px,
  )
  and
  (
    max-width: 575px,
  )
  and
  (
    orientation: landscape,
  ),
  'min-sm-height': (
    min-height: 1024px,
  ),
  'min-sm-medium': (
    min-width: 575px,
  ),
  'min-sm': (
    min-width: 769px,
  ),
  'min-sm-large': (
    min-width: 850px,
  ),
  'min-md': (
    min-width: 992px,
  ),
  'min-md-large': (
    min-width: 1024px,
  ),
  'min-md-large-portrait': (
    min-width: 768px,
  )
  and
  (
    max-width: 1024px,
  )
  and
  (
    orientation: portrait,
  ),
  'min-md-large-landscape': (
    min-width: 768px,
  )
  and
  (
    max-width: 1024px,
  )
  and
  (
    orientation: landscape,
  ),
  'min-lg-small': (
    min-width: 1170px,
  ),
  'min-lg': (
    min-width: 1200px,
  ),
  'min-lg-medium-small': (
    min-width: 1300px,
  ),
  'min-lg-medium': (
    min-width: 1320px,
  ),
  'min-lg-large': (
    min-width: 1440px,
  ),
);

@mixin respond-to($breakpoint) {
  $raw-query: map-get($breakpoints, $breakpoint);

  @if $raw-query {
    $query: if(type-of($raw-query) == 'string', unquote($raw-query), inspect($raw-query));

    @media #{$query} {
      @content;
    }
  } @else {
    @error 'No value found for `#{$breakpoint}`. '
    +'Please make sure it is defined in `$breakpoints` map.';
  }
}

// MIXINS
@mixin display-flex() {
  display: flex;
}

@mixin align-items($position) {
  align-items: $position;
}

@mixin justify-content($position) {
  justify-content: $position;
}

@mixin vertical-align {
  position: relative;
  top: 50%;
  @include translateY(-50%);
}

@mixin flex-center {
  @include display-flex();
  @include align-items(center);
  @include justify-content(center);
}

@mixin flex-center-vertical {
  @include display-flex();
  @include align-items(center);
}

@mixin separator {
  content: '';
  position: absolute;
  top: 0;
  @include size(1px, 100%);
  background-color: $color-base-gray-dark;
}

@mixin background-cover {
  @include background-size(cover);
  background-position: center;
}

@mixin ellipsis {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@mixin no-padding {
  padding: 0;
}

@mixin link {
  font-size: 13px;
  color: $color-primary;
  font-weight: 600;
}
