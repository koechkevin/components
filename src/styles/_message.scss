@import './global';

.message {
  position: relative;

  &:global(.ant-list-item) {
    display: block;
    padding: 8px 32px;

    @include respond-to('sm') {
      padding: 0;
      margin-bottom: 8px;
    }
  }

  :global(.ant-list-item-meta) {
    @include respond-to('sm') {
      padding: 8px 16px;
      overflow: hidden;
    }
  }

  &:hover {
    background: $color-gray-light;
  }

  :global(.ant-list-item-meta-title) {
    margin-bottom: 12px;
  }

  :global(.ant-list-item-meta-content) {
    max-width: calc(100% - 56px);
  }

  :global(.ant-list-item-meta-description) {
    font-size: 13px;
    line-height: 21px;
    color: $color-base-dark;
  }

  .name {
    display: inline-block;
    line-height: 21px;
    font-size: 14px;
    font-weight: bold;
    color: $color-font-darker;
  }

  .created {
    display: inline-block;
    margin-left: 4px;
    height: 21px;
    line-height: 21px;
    font-size: 14px;
    color: $color-base-gray-dark;
  }

  .text {
    font-size: 14px;
    font-weight: normal;
    color: $color-base-dark;
    word-break: break-word;
    white-space: pre-wrap;

    a {
      display: inline-block;
      font-size: 14px;
      font-weight: normal;
    }
  }

  .tag {
    padding-left: 10px;
    font-size: 12px;
    color: $color-base-gray-dark;
  }
}