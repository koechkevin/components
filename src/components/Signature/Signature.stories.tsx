import { select, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';

import { Center } from '../Center';
import { Signature } from '../Signature';

const stories = storiesOf('Signature', module);

stories.addDecorator((story) => <Center>{story()}</Center>);

stories.add('Default', () => {
  const name = text('Name', 'Signature');
  const label = text('Label', 'Label');
  const status = select(
    'Status',
    {
      owner: 'Owner',
      candidate: 'candidate',
    },
    'Owner',
  );
  const visibility = select(
    'visibility',
    {
      default: 'visible',
      hidden: 'hidden',
    },
    'visible',
  );
  const props = { visibility, name, status, label };

  return <Signature {...props} />;
});
