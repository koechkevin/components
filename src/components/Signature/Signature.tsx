import { Avatar as AntdAvatar, Row, Tag as AntdTag } from 'antd';
import React, { FC } from 'react';

import styles from './Signature.module.scss';

export interface LabelProps {
  label?: string;
}

const Label: FC<LabelProps> = (props) => {
  const { label } = props;
  return <AntdTag className={styles.default}>{label}</AntdTag>;
};

export interface AvatarProps {
  nickName?: string;
}

const Avatar: FC<AvatarProps & React.HTMLAttributes<HTMLDivElement>> = (props) => {
  const { nickName } = props;
  return <AntdAvatar className={styles.avatar}>{nickName}</AntdAvatar>;
};

export interface Props {
  visibility: 'visible' | 'hidden';
  name?: string;
  status: string;
  label?: string;
}

export interface ISignatureLabel {
  name?: string;
  visibility: 'visible' | 'hidden';
  style?: React.CSSProperties;
  label?: string;
}

export interface ISignatureAvatar {
  name?: string;
  visibility: 'visible' | 'hidden';
  style?: React.CSSProperties;
}

const SignatureLabel: FC<ISignatureLabel & React.HTMLAttributes<HTMLDivElement>> = (props) => {
  const { name, visibility, label } = props;
  return (
    <AntdTag className={styles.label} style={{ visibility }}>
      {name && <Row className={styles.outerLabel}>{name}</Row>}
      {label && <Label label={label} />}
    </AntdTag>
  );
};

const SignatureAvatar: FC<ISignatureAvatar & React.HTMLAttributes<HTMLDivElement>> = (props) => {
  const { name, visibility } = props;
  const initials = (name && name.match(/\b\w/g)) || [];
  const nickName = ((initials.shift() || '') + (initials.pop() || '')).toUpperCase();

  return (
    <AntdTag className={styles.label} style={{ visibility }}>
      <span className={styles.outerLabel}>{name}</span>
      <Avatar nickName={nickName} />
    </AntdTag>
  );
};

const Signature: FC<Props> = (props) => {
  const { visibility, name, status, label } = props;

  return status === 'Owner' ? (
    <SignatureLabel visibility={visibility} name={name} label={label} className={styles.label} />
  ) : (
    <SignatureAvatar name={name} className={styles.label} visibility={visibility} />
  );
};

export default Signature;
