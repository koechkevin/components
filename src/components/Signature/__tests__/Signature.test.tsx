import { Tag as AntdTag } from 'antd';
import { shallow } from 'enzyme';
import React from 'react';

import Signature from '../Signature';
import styles from './Signature.module.scss';

describe('component App', () => {
  it('component render without props without failing', () => {
    const component = shallow(<Signature />);
    expect(component).toMatchSnapshot();
  });

  it('component render with props without failing', () => {
    const propsForTest = {
      visibility: true,
      nameLabel: 'name',
    };
    const component = shallow(<Signature {...propsForTest} />);
    expect(component).toMatchSnapshot();
  });

  // loading - true
  it(' component has styling', () => {
    interface Props {
      children: React.ReactNode;
    }
    const styleSignature = { display: 'flex' } as any;
    const Signature: FC<Props> = ({ children }) => <AntdTag>{children}</AntdTag>;
    const nameLabel = 'Name';
    const tree = shallow(
      <wrapper>
        <Signature className={styles.label} />
      </wrapper>,
    );

    expect(tree).toMatchInlineSnapshot(`ShallowWrapper {}`);
  });
});
