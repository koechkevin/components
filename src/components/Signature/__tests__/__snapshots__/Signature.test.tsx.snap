// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component App component render with props without failing 1`] = `ShallowWrapper {}`;

exports[`component App component render without props without failing 1`] = `ShallowWrapper {}`;
